gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowTriggerHandler.java

```diff
@@ -313,4 +313,46 @@ protected static long getUTCTimeFromDelayPeriod(long delayPeriodMillis) {
     Date date = Date.from(localDateTime.atZone(ZoneId.of("UTC")).toInstant());
     return GobblinServiceJobScheduler.asUTCEpochMillis(date);
   }
+
+  /**
+   * Attempts to acquire lease for a given {@link DagTask} through lease arbitration
+   * @param dagTask for which we want to attempt to acquire a lease
+   * @return true if successfully acquired lease, or else false
```

Comment:
as mentioned already, the `DagTask` must hold on to its specific `LeaseObtainedStatus`